src/Microsoft.AspNet.Mvc.TagHelpers/Internal/MissingAttributeLoggerStructure.cs

```diff
@@ -24,15 +24,19 @@ public class MissingAttributeLoggerStructure : ILoggerStructure
         /// </summary>
         /// <param name="uniqueId">The unique ID of the HTML element this message applies to.</param>
         /// <param name="missingAttributes">The missing required attributes.</param>
-        public MissingAttributeLoggerStructure(string uniqueId, IEnumerable<string> missingAttributes)
+        /// <param name="extraValues">Extra values to include in the log structure.</param>
+        public MissingAttributeLoggerStructure(
+            string uniqueId,
+            IEnumerable<string> missingAttributes)
         {
             _uniqueId = uniqueId;
             MissingAttributes = missingAttributes;
-            _values = new Dictionary<string, object>
+            var values = new Dictionary<string, object>
             {
-                { "UniqueId", _uniqueId },
-                { "MissingAttributes", MissingAttributes }
+                ["UniqueId"] = _uniqueId,
+                ["MissingAttributes"] = MissingAttributes
             };
+            _values = values;
```

Comment:
I think that's a carry over from when this ctor accepted extra values (I'd create the default values then add the ones passed in). Will change it to just directly assign to `_values`.